src/advanced_inv.cpp

```diff
@@ -2190,45 +2238,49 @@ void advanced_inventory::draw_minimap()
     }
 }
 
-bool advanced_inventory::set_vehicle(aim_location sel)
+bool advanced_inventory::set_vehicle(advanced_inventory_pane &pane, aim_location sel)
 {
     // allow if dragged, otherwise only stick to directions that can store in vehicle cargo
     if(sel > AIM_DRAGGED || sel < AIM_SOUTHWEST || !squares[sel].can_store_in_vehicle()) {
+        swap_panes();
         const char *msg = nullptr;
         // throw in some flavour :^)
         switch(sel) {
-            case AIM_INVENTORY:
-                msg = _("You got a car in your pocket?");
-                break;
-            case AIM_WORN:
-                msg = _("Don't know how you'd wear a truck...");
-                break;
-            case AIM_ALL:
-                msg = _("Pick a specific tile! Don't be TOO antsy!");
+            case NUM_AIM_LOCATIONS:
+                msg = _("This here be a bug! [NUM_AIM_LOCATIONS] in set_vehicle()");
                 break;
             case AIM_CONTAINER:
                 msg = _("In the Cataclysm, it's doubtful you will find a ship in a bottle...");
                 break;
             case AIM_VEHICLE:
-                msg = _("Your thoughts drift to a dream within a dream...");
+                swap_panes();
                 break;
             default:
-                msg = (squares[sel].veh != nullptr) ?
-                    _("Where ya gonna put it, genius?") :
-                    _("Wishing for a Ferrari won't make it happen...");
+                msg = (squares[sel].veh != nullptr) ? 
+                    _("There isn't a cargo area there!") : 
+                    _("There isn't a vehicle there!");
                 break;
         }
-        popup(msg);
+        if(msg != nullptr) {
+            popup(msg);
+        }
         return false;
     }
+    // note the location the vehicle is being set to, per pane
+    auto loc = squares[sel].offset_to_location();
+    pane.veh_area = loc;
     // setup on the square's offsets, as a second safeguard
-    auto &valid_square = squares[squares[sel].offset_to_location()];
+    auto &valid_square = squares[loc];
     squares[AIM_VEHICLE].set_vehicle(valid_square);
     return true;
 }
 
 void advanced_inv_area::set_vehicle(advanced_inv_area &s)
 {
+    if(veh == nullptr) {
```

Comment:
Well, you call this function from the other `set_vehicle` function on `squares[AIM_VEHICLE]`, which has been initialized (see `advanced_inv_area::init`) to have `veh == nullptr` (the case for `AIM_VEHICLE` does not set `veh` to anything). As a result, this test is usually true, especially when re-entering the AIM.